feat: Contextualize prompts and search with current date and time

ngoiyaeric opened the pull request:
### **User description**
- I updated `lib/agents/researcher.tsx` to ensure the system prompt includes the current date and time, formatted as YYYY-MM-DD HH:MM:SS.
- I modified `lib/agents/tools/search.tsx` to append the current date and time (YYYY-MM-DD HH:MM:SS) to the Tavily search query, aiming for more time-sensitive results.
- I added unit tests for these changes. Note: Tests were written but could not be executed as no testing framework is currently configured in the project.


___

### **PR Type**
Enhancement, Tests


___

### **Description**
• Add current date/time context to researcher system prompts
• Enhance search queries with timestamp for time-sensitive results
• Implement comprehensive unit tests for date formatting functionality
• Export tavilySearch function for improved testability


___



### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>researcher.test.tsx</strong><dd><code>Add unit tests for researcher date formatting</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

lib/agents/researcher.test.tsx

• Add comprehensive unit tests for researcher agent<br> • Mock Date object <br>to test consistent date formatting<br> • Verify system prompt includes <br>correctly formatted current date<br> • Test date format as "YYYY-MM-DD <br>HH:MM:SS"


</details>


  </td>
  <td><a href="https://github.com/QueueLab/QCX/pull/193/files#diff-c2fe5e560dbaefc17eaae3a95d811070c44ecb6fca8a706d0a6245d77de3bffb">+129/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>search.test.tsx</strong><dd><code>Add unit tests for search date enhancement</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

lib/agents/tools/search.test.tsx

• Add unit tests for search tool date functionality<br> • Mock Date object <br>and fetch API for testing<br> • Verify tavilySearch appends formatted date <br>to queries<br> • Test date format consistency in search requests


</details>


  </td>
  <td><a href="https://github.com/QueueLab/QCX/pull/193/files#diff-1bbbd960d3a03540d69992f30d4469d2de4505244cba389e3da5d9e9687cad00">+100/-0</a>&nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>researcher.tsx</strong><dd><code>Implement standardized date formatting for prompts</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

lib/agents/researcher.tsx

• Replace <code>toLocaleString()</code> with custom date formatting<br> • Format <br>current date as "YYYY-MM-DD HH:MM:SS"<br> • Include formatted date in <br>system prompt context


</details>


  </td>
  <td><a href="https://github.com/QueueLab/QCX/pull/193/files#diff-07a588855c7c4f7ac8f731bb6b4876aa2fa6e9fda6096e999d0c131a3444a251">+8/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>search.tsx</strong><dd><code>Export tavilySearch and add date context</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

lib/agents/tools/search.tsx

• Export tavilySearch function for testability<br> • Add current date/time <br>to search queries for context


</details>


  </td>
  <td><a href="https://github.com/QueueLab/QCX/pull/193/files#diff-c143ac4c60674a64c34031f829803aafdf41345cb92b39b54bb3e1af90547262">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about Qodo Merge usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

google-labs-jules[bot] committed:
feat: Contextualize prompts and search with current date and time

- I updated `lib/agents/researcher.tsx` to ensure the system prompt includes the current date and time, formatted as YYYY-MM-DD HH:MM:SS.
- I modified `lib/agents/tools/search.tsx` to append the current date and time (YYYY-MM-DD HH:MM:SS) to the Tavily search query, aiming for more time-sensitive results.
- I added unit tests for these changes. Note: Tests were written but could not be executed as no testing framework is currently configured in the project.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/QueueLab/QCX?pullRequest=193) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/QueueLab/QCX?pullRequest=193) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/QueueLab/QCX?pullRequest=193) it.</sub>

google-labs-jules[bot] committed:
feat: Implement location-aware query processing

This feature enables the application to understand when your query
requires your current location, simulate a permission request flow,
and use this location (if notionally granted) to provide more
contextual results from search and geospatial tools.

Key changes:

- **Researcher (`lib/agents/researcher.tsx`):**
    - Introduced a capability to ask for your location when a query depends on it.
    - I implemented logic to simulate a location permission flow (50/50 grant/deny for now).
    - Based on the simulated outcome, I re-engage with either your location (currently hardcoded to San Francisco for simulation if granted) or a notification of denial.
    - If location is "granted", `currentUserLocation` is passed to `getTools`.
    - My system prompts were updated to guide me on using this new capability,
      interpreting location/denial messages, and leveraging location
      with other tools.

- **Tools (`lib/agents/tools/index.tsx`, `search.tsx`, `geospatial.tsx`):**
    - `ToolProps` updated to accept `currentUserLocation`.
    - `searchTool`: If location is available and query isn't already
      specific, appends "in [place_name]" to the search query.
    - `geospatialTool`: If location is available and query is generic,
      appends "near [place_name]" to the query.

- **Testing:**
    - Unit test logic written (but not executed due to lack of a test
      framework) for `researcher.tsx` covering location request
      initiation, handling of simulated grant/denial, and assertion
      of query augmentation for tools.

Note: The UI components for actual browser location permission requests
and state management (`LocationContext`) are not part of this commit
and will need to be implemented separately to complete the end-to-end flow.
The current implementation provides the backend logic to support this.

google-labs-jules[bot] committed:
fix: Address no response for direct time queries

This commit fixes an issue where querying for the current time (e.g., "what is the time?") could result in no visible response if I incorrectly attempted to determine your location as the sole action.

Changes:

1.  **`lib/agents/researcher.tsx`**:
    - I added logic after the main response processing loop to check if I was trying to determine your location AND I provided no textual output.
    - If both conditions are met, a holding message ("Your request is being processed, and I may need to confirm location preferences. Please wait a moment.") is now set as the full response and streamed to you. This ensures you receive feedback instead of a blank response while the (simulated) location flow proceeds.

2.  **`lib/agents/researcher.tsx` (System Prompt)**:
    - I made a minor refinement to my internal instructions. The line providing the current date and time now also includes: "You can use this information directly to answer questions about the current time or date." This gently encourages me to use the provided context for such queries.

These changes ensure a better experience for you by providing immediate feedback for the affected queries and make me slightly less likely to misinterpret direct time/date questions.
